src/main/kotlin/com/exactpro/th2/common/ConfigurationProviderFactory.kt

```diff
@@ -35,27 +32,15 @@ interface ConfigurationProviderFactory {
      * Method for instantiating a [ConfigurationProvider] using parameters.
      * Each factory implementation can have its own arguments
      *
-     * To get an array of arguments, use method [parseCommandLine]
-     */
-    fun createProvider(args: Array<String>): ConfigurationProvider
-
-    /**
-     * To parse command line arguments using the apache.commons.cli library,
-     * you must add possible arguments to the [Options] object.
-     * Otherwise, the parser will encounter unknown arguments and throw an exception.
-     *
-     * Since each factory has its own arguments,
-     * this is method for adding arguments to [Options].
+     * To get an array of arguments, use method [parseArguments]
      */
-    fun addOwnOptionsToCmd(options: Options)
+    fun create(args: Array<String>): ConfigurationProvider
 
     /**
+     * The factory is responsible for selecting the required parameters from the whole list.
      * This method allows you to parse the command line object
      * to get the right parameters in the right order.
-     *
-     * Make sure you called [addOwnOptionsToCmd] before parsing [CommandLine] object.
-     * An exception will be thrown otherwise.
      */
-    fun parseCommandLine(cmd: CommandLine): Array<String>
+    fun parseArguments(args: Array<String>): Array<String>
```

Comment:
Why didn't we use the CommandLine that provides methods for getting options by their names? Currently, each factory has to work with a raw array and has to determine the right order

Reply:
If we make a method for creating a provider from CommandLine, such a method will perform several actions at once. In addition, there are other constructors in the CommonFactory, for example, from the Settings class. The "provider name + arguments" couple can be easily stored in Settings, passed directly to constructors and so on

Reply:
Ok, I see, you mean a method that just get Array<String> from CommandLine. Well, I'll give it a try